### PR TITLE
Zycommon refactor

### PR DESCRIPTION
**Pull request by JakeFrey**
Review for refactoring and using zycommon and adding in a last few changes to wrap up the demo app

### CODE REVIEW

File: ' [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import Ember from 'ember';
```
Comment (ssirowy) [inline ']: Remove this file. Looks like a stray.

File: app/controllers/sign-in.js [hunk 2]
```diff
@@ -1,32 +1,22 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
-    cuser: Ember.inject.service('current-user-service'),
+    session: Ember.inject.service('session'),
```
Comment (ssirowy) [inline app/controllers/sign-in.js]: When the service name is the same name as the property, you can use a shorthand like this:
```  javascript
session: Ember.inject.service(),
```

File: app/controllers/sign-in.js [hunk 3]
```diff
@@ -1,32 +1,22 @@
...
-                    else {
-                        alert('Something\'s wrong');
-                    }
-                },
-                error: function(request, textStatus, errorThrown) {
-                    alert(request + textStatus + errorThrown + 'an error');
+            this.get('session').authenticate(loginParams).then(
+                () => {
+                    this.transitionToRoute('zybooks');
+                    debugger;
```
Comment (ssirowy) [inline app/controllers/sign-in.js]: Remove debugger.

File: app/controllers/sign-in.js [hunk 4]
```diff
@@ -1,32 +1,22 @@
...
                 },
-            });
+                error => {
+                    debugger;
```
Comment (ssirowy) [inline app/controllers/sign-in.js]: Remove debugger.  You probably also want to make use of this error if the credentials are incorrect.

File: app/routes/zybooks.js [hunk 5]
```diff
@@ -1,34 +1,9 @@
 import Ember from 'ember';
-import Zybook from '../models/zybook';
 
 export default Ember.Route.extend({
-    cuser: Ember.inject.service('current-user-service'),
+    session: Ember.inject.service('session'),
```
Comment (ssirowy) [inline app/routes/zybooks.js]: Same here.

File: app/services/session.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+    user: null,
+    credential: '',
+    zyserver: Ember.inject.service('zyserver'),
```
Comment (ssirowy) [inline app/services/session.js]: Same here.

**Comment by ssirowy** [comment]
Just a few comments @JakeFrey

**Comment by JakeFrey** [comment]
I made a push to hopefully fix all the changes you mentioned. I also finished the log out button, fixed the table of contents when opening a chapter, and have error messages displayed in the form when trying to log in. Let me know if there's anything else I can add to enhance it that you think would be good practice for me.

File: app/routes/zybooks.js [hunk 7]
```diff
@@ -1,34 +1,16 @@
 import Ember from 'ember';
-import Zybook from '../models/zybook';
 
 export default Ember.Route.extend({
-    cuser: Ember.inject.service('current-user-service'),
+    session: Ember.inject.service(),
 
+    beforeModel: function(transition) {
+        debugger;
```
Comment (ssirowy) [inline app/routes/zybooks.js]: Remove debugger;

File: app/services/session.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+    user: null,
+    credential: '',
```
Comment (ssirowy) [inline app/services/session.js]: Credential should probably default to null.

File: app/styles/app.scss [hunk 9]
```diff
@@ -41,11 +41,23 @@ body > .ember-view {
...
 }
 
+#logout-button {
+    color: white;
+    position: absolute;
+    right: 30px;
+    top: -5px;
+}
+
+.hide {
```
Comment (ssirowy) [inline app/styles/app.scss]: Didnt you come up with a solution that didnt involved this class?

File: app/controllers/zybook.js [hunk 10]
```diff
@@ -1,4 +1,13 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
+    reset: function() {
+        const chapters = this.get('content.chapters');
+        
+        chapters.forEach(function(item, index) {
+            if (index !== chapters.content.length - 1) {
+                item.sections[item.sections.length - 1].last = true;
```
Comment (ssirowy) [inline app/controllers/zybook.js]: Use `.set` syntax.

**Comment by ssirowy** [comment]
Nice work @JakeFrey

**Pull request merged**